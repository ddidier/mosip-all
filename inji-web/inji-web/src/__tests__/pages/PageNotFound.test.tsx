import React from 'react';
import { screen, fireEvent } from '@testing-library/react';
import { PageNotFound } from '../../pages/PageNotFound';
import { renderWithRouter,mockusei18n} from '../../test-utils/mockUtils';

mockusei18n();
//todo : extract the local method to mockUtils, which is added to bypass the routing problems
// Mock the useNavigate hook
const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

describe('Testing the Layout of PageNotFound', () => {
  test('Check the presence of the PageNotFound container', () => {
    const { asFragment } = renderWithRouter(<PageNotFound />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Testing the Functionality of PageNotFound', () => {
  test('Check if clicking the home button navigates to the home page', () => {
    renderWithRouter(<PageNotFound />);
    fireEvent.click(screen.getByTestId('PageNotFound-Home-Button'));
    expect(mockNavigate).toHaveBeenCalledWith('/');
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
