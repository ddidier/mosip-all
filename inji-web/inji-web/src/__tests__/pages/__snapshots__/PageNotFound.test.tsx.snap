// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Layout of PageNotFound Check the presence of the PageNotFound container 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center items-center h-full flex-col"
  >
    <div>
      <strong>
        404.
      </strong>
      That's an error
    </div>
    <h3
      class="pt-4"
    >
      The requested URL / was not found on this server.
    </h3>
    <div
      class="p-8"
    >
      <button
        class="text-iw-primary font-bold py-2 px-4 rounded-lg border-2 border-iw-primary"
        data-testid="PageNotFound-Home-Button"
      >
        Go To Home
      </button>
    </div>
  </div>
</DocumentFragment>
`;
